pkg/controller/machinepool/azureactuator.go

```diff
@@ -101,17 +105,14 @@ func (a *AzureActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *
 		if err != nil {
 			return nil, false, err
 		}
-		if gen2ImageExists {
-			// Get HyperV Generation to provide to installer machineset generation. The HyperV Generation is germane to the instance/disk
-			// type and affects the image used for the instance. "-gen-2" will be appended to the image name when the
-			// hyperVGen == "V2".
-			hyperVGen, err = a.client.GetHyperVGenerationVersion(context.TODO(),
-				computePool.Platform.Azure.InstanceType,
-				computePool.Platform.Azure.OSDisk.DiskType,
-				cd.Spec.Platform.Azure.Region,
-			)
-			if err != nil {
-				return nil, false, err
+		if !gen2ImageExists {
+			// Modify capabilities to ensure that a V1 image is chosen by installazure.MachineSets
+			// because a gen2 image does not exist.
+			// The HyperVGeneration is germane to the instance/disk type and affects the image used
+			// for the instance. "-gen-2" will be appended to the image name when HyperVGenerations
+			// capability includes "V2".
```

Comment:
I think this piece of commentary was left over from a previous iteration where we were string scraping to figure out the version?

Reply:
The `installer.MachineSets()` call below will append `"-gen2"` to the image name based on `"HyperVGenerations"`  within the `capabilities` map. I'll make that more clear in the comment.